### PR TITLE
Avoid crashing on using the index.lifecycle.name in the API body

### LINKED ISSUE #1019: Open-search process crashing on using the index.lifecycle.name in the API body #102
**Describe the bug**
While applying the policy _index_template on an index, Open-search crashed on using the index.lifecycle.name in the API body. This setting is not available in open-search but it shouldn't crash the process and return with the exception.

This is occurring on a vanilla OS cluster even without index management plugin installed on the cluster. 

Issue opened in Index Management Repo: https://github.com/opensearch-project/index-management/issues/102

**To Reproduce**
* Run an open-search cluster. 
* Hit the endpoint with an API: `curl -X PUT "localhost:9200/_index_template/timeseries_template?pretty" -H 'Content-Type: application/json' -d'
{
  "index_patterns": ["*"],
  "data_stream": { },
  "template": {
    "settings": {
      "number_of_shards": 1,
      "number_of_replicas": 1,
      "index.lifecycle.name": "timeseries_policy"
    }
  }
}
'
`
* The response returned is `curl: (52) Empty reply from server`
* The log trace found in the ES logs is pasted below. 

**Expected behavior**
As the setting is not available the process shouldn't crash but return the exception as setting not found.

**Log Trace**
`
=== Standard error of node `node{::runTask-0}` ===
»   ↓ last 40 non error or warning messages from /local/home/aditjind/gh/OpenSearch/build/testclusters/runTask-0/logs/opensearch.stderr.log ↓
» fatal error in thread [opensearch[runTask-0][masterService#updateTask][T#1]], exiting
»  java.lang.AssertionError: expected [index.lifecycle.name] to be private but it was not
»       at org.opensearch.cluster.metadata.MetadataCreateIndexService.validatePrivateSettingsNotExplicitlySet(MetadataCreateIndexService.java:1047)
»       at org.opensearch.cluster.metadata.MetadataCreateIndexService.getIndexSettingsValidationErrors(MetadataCreateIndexService.java:1037)
»       at org.opensearch.cluster.metadata.MetadataIndexTemplateService.validate(MetadataIndexTemplateService.java:1305)
»       at org.opensearch.cluster.metadata.MetadataIndexTemplateService.validate(MetadataIndexTemplateService.java:1226)
»       at org.opensearch.cluster.metadata.MetadataIndexTemplateService.validate(MetadataIndexTemplateService.java:1221)
»       at org.opensearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:507)
»       at org.opensearch.cluster.metadata.MetadataIndexTemplateService$4.execute(MetadataIndexTemplateService.java:403)
»       at org.opensearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:60)
»       at org.opensearch.cluster.service.MasterService.executeTasks(MasterService.java:715)
»       at org.opensearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:337)
»       at org.opensearch.cluster.service.MasterService.runTasks(MasterService.java:232)
»       at org.opensearch.cluster.service.MasterService.access$000(MasterService.java:86)
»       at org.opensearch.cluster.service.MasterService$Batcher.run(MasterService.java:164)
»       at org.opensearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:163)
»       at org.opensearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:201)
»       at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:697)
»       at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedOpenSearchThreadPoolExecutor.java:266)
»       at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedOpenSearchThreadPoolExecutor.java:229)
»       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
»       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
»       at java.base/java.lang.Thread.run(Thread.java:832)`

**Host/Environment (please complete the following information):**
 - OS: Linux
 - Version: 5.4.116-64.217.amzn2int.x86_64

### PR DESCRIPTION
**Pull request by frotsch**
Signed-off-by: frotsch <frotsch@mailbox.org>

### Description
Fix OpenSearch process crashing on using the index.lifecycle.name in the API body when asserts are enabled
 
### Issues Resolved
#1019
 
### Check List
- [ x] All tests pass
- [x ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java [hunk 1]
```diff
@@ -1044,7 +1044,10 @@ public void validateIndexSettings(String indexName, final Settings settings, fin
         for (final String key : settings.keySet()) {
             final Setting<?> setting = indexScopedSettings.get(key);
             if (setting == null) {
-                assert indexScopedSettings.isPrivateSetting(key) : "expected [" + key + "] to be private but it was not";
+                //https://github.com/opensearch-project/OpenSearch/issues/1019
```
Comment (dblock) [inline server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java:1047]: I see a lot of "// see: [github link]" in the code, maybe we can just adopt that "convention"

The fact that there's no space after `//` is really hurting my OCD ;)

Comment (frotsch) [reply server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java]: Done

**Comment by dblock** [comment]
start gradle check

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Merged, could you please backport PR this one into 1.1 @frotsch?

**Comment by frotsch** [comment]
How exactly can I backport the PR? @dblock

**Comment by dblock** [comment]
Make a new branch off `1.x`, cherry pick the merged commit on it and make a PR into 1.x. Thanks!

**Pull request merged**